sympy/physics/mechanics/_geometry.py

```diff
@@ -0,0 +1,571 @@
+        >>> pO = Point('pO')
+        >>> pO.set_vel(N, 0)
+        >>> sphere = Sphere(r, pO)
+        >>> p1 = Point('p1')
+        >>> p2 = Point('p2')
+
+        Let's assume that ``p1`` lies at a distance of ``r`` in the ``N.x``
+        direction from ``pO`` and that ``p2`` is located on the sphere's
+        surface in the ``N.y + N.z`` direction from ``pO``. These positions can
+        be set with:
+
+        >>> p1.set_pos(pO, r * N.x)
+        >>> p1.pos_from(pO)
+        r*N.x
+        >>> p2.set_pos(pO, r * (N.y + N.z).normalize())
+        >>> p2.pos_from(pO)
+        sqrt(2)*r/2*N.y + sqrt(2)*r/2*N.z
+
+        The geodesic length, which is in this case a quarter of the sphere's
+        circumference, can be calculated using the ``geodesic_length`` method:
+
+        >>> sphere.geodesic_length(p1, p2)
+        pi*r/2
+
+        If the ``geodesic_length`` method is passed an argument ``Point`` that
+        doesn't lie on the sphere's surface then a ``ValueError`` is raised
+        because it's not possible to calculate a value in this case.
+
+        Parameters
+        ==========
+
+        point_1 : Point
+            The point from which the geodesic length should be calculated.
+        point_2 : Point
+            The point to which the geodesic length should be calculated.
+
+        """
+        for point in (point_1, point_2):
+            if not self._point_is_on_surface(point):
+                msg = (
+                    f'Geodesic length cannot be calculated as point {point} '
+                    f'with radius {point.pos_from(self.point).magnitude()} '
+                    f'from the sphere\'s center {self.point} does not lie on '
+                    f'the surface of {self} with radius {self.radius}.'
+                )
+                raise ValueError(msg)
+        point_1_vector = point_1.pos_from(self.point).normalize()
+        point_2_vector = point_2.pos_from(self.point).normalize()
+        central_angle = acos(cancel(point_2_vector.dot(point_1_vector)))
+        geodesic_length = self.radius * central_angle
+        return geodesic_length
+
+    def __repr__(self) -> str:
+        """Representation of a ``Sphere``."""
+        return (
+            f'{self.__class__.__name__}(radius={self.radius}, '
+            f'point={self.point})'
+        )
+
+
+class Cylinder(GeometryBase):
+    """A solid (infinite) cylindrical object.
+
+    Explanation
+    ===========
+
+    A wrapping geometry that allows for circular arcs to be defined between
+    pairs of points. These paths are always geodetic (the shortest possible) in
+    the sense that they will be a straight line on the unwrapped cylinder's
+    surface. However, it is also possible for a direction to be specified, i.e.
+    path's can be influenced such that they either wrap along the shortest side
+    or the longest side of the cylinder. To define these directions, the
+    right-hand screw rule is used, i.e. looking along the axis, the path will
```

Comment:
This might be flavoured UK: I have generally heard this only as "right-hand rule" but found the "screw" on a UK page. 

When you say "looking along the axis" and cw, it defeats the purpose of citing the rule. I like the UK page (except for reference to screw) in that it avoid "looking" and "cw" since those are a function of which direction you look. It says, "The axis is 'grasped' in the right hand, the fingers curl round in the direction of positive rotation and the thumb is orientated in the positive direction." So we should just say something like "rotation is in the positive direction following the right-hand rule" OR forget reference to the rule and say that "the rotation is cw when looking along the axis in the positive direction" -- note that "positive" gives the direction that one is looking, e.g. standing on origin and looking towards (1,0,0).

Reply:
I was taught using the screw rule (in the UK).

The "right hand rule" (RHR) and the "right hand screw" (RHS) are not the same thing although they conventionally go together. The RHR defines the direction of a cross-product in physical (not mathematical) space. Equivalently you can say that the RHR defines the positive direction of the third axis of a Cartesian coordinate system for a 3D physical space (since the standard unit vectors are conventionally related like `i x j = k`).

The RHS defines the interpretation of some sense of rotation that is associated with the "direction" of a pseudovector again in physical space but "the rule" is only used to specify the physical interpretation of a particular quantity e.g. angular velocity or magnetic field direction etc. For example we can define the direction of the angular velocity `w` of a rotating body by saying that if the right-hand thumb points in the direction of `w` then the curled fingers point towards the sense of rotation (this statement can hold without reference to any cross products or coordinate systems that might be left or right handed).

Here it makes sense to reference the screw rule because you are trying to specify the orientation of a rotation associated with a given vector.

Where there are equations involving both cross products and pseudovectors which combination of right/left hand rule and right/left hand screw is understood determines the order of the cross products. The equations typically shown in mechanics like `L = r x p` etc are based around either using both RHR and RHS together *or* using both LHR and LHS together. In this case the RHR (or LHR) determines the physical direction of the cross product but the RHS (or LHS) determines the physical interpretation of what the physical direction of `L` is supposed to mean. Mixing and matching like RHR and LHS means that the cross product needs to be reversed like `L = p x r` but otherwise the equations are no different under the left or right handed versions.

In that sense it is not really meaningful to say here that the right hand rule is used any more than the left hand rule although we can say that for a left or right handed coordinate system the corresponding screw rule applies.

I'm not suggesting that the docstring should attempt to explain this though. If there is an implied sense that all coordinate systems in the mechanics module use the right hand rule then it can be informally meaningful to say that the direction here is also determined by "the right hand rule". Whether or not mechanics uses the right hand rule for its coordinate systems is only meaningful when applied to 3D visualisations though because all equations and numeric values would be the same either way.